tests/test_frompyobject.rs

```diff
@@ -0,0 +1,308 @@
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyString, PyTuple};
+use pyo3::{PyErrValue, PyMappingProtocol};
+
+#[macro_use]
+mod common;
+
+#[derive(Debug, FromPyObject)]
+pub struct A<'a> {
+    #[extract(getattr)]
```

Comment:
Considering the possibility of a to-python conversion later, I'd prefer use a term that could be symmetric. e.g. `#[pyo3(field_type = "attribute")]`, or maybe just `#[pyo3(attribute)]`